miasm/os_dep/linux/syscall.py

```diff
@@ -169,14 +176,22 @@ def sys_x86_32_socket(jitter, linux_env):
         #           socklen_t addrlen);
         fd = jitter.vm.get_u32(jitter.cpu.ESP)
         socklen = jitter.vm.get_u32(jitter.cpu.ESP+8)
-        # Not the exact size because shellcodes won't provide the full struct
-        sockaddr = jitter.vm.get_mem(jitter.vm.get_u32(jitter.cpu.ESP+4), 8)
+        try:
```

Comment:
What about using the `socklen` instead of a fixed length?

Reply:
In some cases, a shellcode would not have a full sockaddr struct but just the needed fields, what I have done in the next commit is getting the socklen and if it fails, I only get the first 8 bytes. Is that an ok trick to have it work ?

Reply:
In fact, maybe we should behave like the kernel does so we will be close to a real environment.
If the kernel is ok with semi structures, maybe your patch is ok.

Reply:
I have added a first get_mem for the full structure and fallback to 8 bytes if the memory if not that large, it should be close to what the kernel is doing I guess